### PR TITLE
MISUV-1718 - Added TXM audit event for What You Owe page

### PR DESCRIPTION
**Pull request by Wright93**

### CODE REVIEW

File: it/controllers/PaymentDueControllerISpec.scala [hunk 1]
```diff
@@ -37,6 +39,7 @@ class PaymentDueControllerISpec extends ComponentSpecBase {
     "Authorised" should {
       "NewFinancialDetailsApi FS is disabled" when {
```
Comment (MihailsC-hmrc) [inline it/controllers/PaymentDueControllerISpec.scala:40]: Optional suggestion: swap `should` with `when`:
```Scala
    "Authorised" when {
        "NewFinancialDetailsApi FS is disabled" should {
```

File: it/controllers/agent/nextPaymentDue/PaymentDueControllerISpec.scala [hunk 2]
```diff
@@ -116,6 +118,8 @@ class PaymentDueControllerISpec extends ComponentSpecBase with FeatureSwitching
       enable(Payment)
       stubAuthorisedAgentUser(authorised = true)
 
+      WhatYouOweChargesList()
+
```
Comment (MihailsC-hmrc) [inline it/controllers/agent/nextPaymentDue/PaymentDueControllerISpec.scala:122]: This change is not needed. Please remove.

File: it/helpers/ComponentSpecBase.scala [hunk 3]
```diff
@@ -88,7 +89,7 @@ trait ComponentSpecBase extends TestSuite with CustomMatchers
 
   override def beforeEach(): Unit = {
     super.beforeEach()
-    wireMockServer.resetMappings()
+    WireMock.reset()
```
Comment (MihailsC-hmrc) [inline it/helpers/ComponentSpecBase.scala:92]: Why this change in a test base class? What's the issue with instance of `wireMockServer`? Why using "default" `WireMock` management?

File: test/audit/models/WhatYouOweResponseAuditModelSpec.scala [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+          Json.obj(
+            "chargeType" -> "Payment on account 1 of 2",
+            "dueDate" -> dueDateInFuture,
+            "outstandingAmount" -> 50
+          ),
+          Json.obj("accruingInterest" -> 12.67,
+            "chargeType" -> "Remaining balance",
+            "dueDate" -> outStandingCharges,
+            "outstandingAmount" -> 123456.67
+          )
```
Comment (MihailsC-hmrc) [inline test/audit/models/WhatYouOweResponseAuditModelSpec.scala:78]: Note: there's no `accruingInterest` field in expected json i.e. it's not tested.

File: test/controllers/PaymentDueControllerSpec.scala [hunk 5]
```diff
@@ -160,10 +170,15 @@ class PaymentDueControllerSpec extends MockAuthenticationPredicate
           when(paymentDueService.getWhatYouOweChargesList()(any(), any()))
             .thenReturn(Future.successful(whatYouOweChargesListEmpty))
 
-          val result = await(controller.viewPaymentsDue(fakeRequestWithActiveSession))
+          val result = await(controller.viewPaymentsDue()(fakeRequestWithActiveSession))
```
Comment (MihailsC-hmrc) [inline test/controllers/PaymentDueControllerSpec.scala:173]: Note: this change is not needed. Please remove it.

**Review by MihailsC-hmrc — CHANGES_REQUESTED** [review]
Please address at least non-optional comments. Please answer the questions.

File: it/controllers/CalculationControllerISpec.scala [hunk 6]
```diff
@@ -440,7 +440,6 @@ class CalculationControllerISpec extends ComponentSpecBase with FeatureSwitching
         IndividualCalculationStub.verifyGetCalculation(testNino, "idOne")
 
         verifyAuditContainsDetail(ReportDeadlinesRequestAuditModel(testUser).detail)
-        verifyAuditContainsDetail(ReportDeadlinesResponseAuditModel(testUser, "ABC123456789", Nil).detail)
```
[the remark this answers is not in the document]
Comment (JSimmons072019) [reply it/controllers/CalculationControllerISpec.scala:443]: when broken tests were fixed this test failed after inspection the audit call couldn't have been made so previously was showing false positive

File: it/helpers/ComponentSpecBase.scala (hunk 3 above)
Comment (JSimmons072019) [reply it/helpers/ComponentSpecBase.scala:92]: wireMockServer.resetMappings() failed to properly reset systems in between tests causing false positives and false negatives through-out the tests

File: test/audit/models/WhatYouOweResponseAuditModelSpec.scala (hunk 4 above)
Comment (JSimmons072019) [reply test/audit/models/WhatYouOweResponseAuditModelSpec.scala:78]: ? line 74 has accruing interest, stubbed test data also has aci data

Comment (MihailsC-hmrc) [reply test/audit/models/WhatYouOweResponseAuditModelSpec.scala:78]: @JSimmons072019 my bad indeed, was blind. No issues then with it.

File: it/helpers/ComponentSpecBase.scala (hunk 3 above)
Comment (MihailsC-hmrc) [reply it/helpers/ComponentSpecBase.scala:92]: Thanks, okay. Think about creating a tech debt to investigate and fix it while this looks-like-workaround solution is in place and does its job.

File: test/audit/models/WhatYouOweResponseAuditModelSpec.scala (hunk 4 above)
Comment (JSimmons072019) [reply test/audit/models/WhatYouOweResponseAuditModelSpec.scala:78]: no problem it was a little hidden :)

File: app/audit/Utilities.scala [hunk 7]
```diff
@@ -19,6 +19,8 @@ package audit
 import auth.MtdItUser
 import play.api.libs.json.{JsObject, Json}
 import utils.Utilities.JsonUtil
+import _root_.models.financialDetails.Charge
```
Comment (MihailsC-hmrc) [inline app/audit/Utilities.scala:22]: Just a note: this automagically added `_root_` package is a bit ugly, usually devs try to get rid of it.

**Review by MihailsC-hmrc — APPROVED** [review]

File: app/audit/Utilities.scala (hunk 7 above)
Comment (MihailsC-hmrc) [reply app/audit/Utilities.scala:22]: Actually, I don't see an easy way to distinguish packages without renamings. Thus, let it be.
